src/key.h

```diff
@@ -22,6 +25,14 @@
  */
 typedef std::vector<unsigned char, secure_allocator<unsigned char> > CPrivKey;
 
+constexpr static size_t ECDH_SECRET_SIZE = CSHA256::OUTPUT_SIZE;
+
+// SHA256("secp256k1_ellsq_xonly_ecdh") proven with a unit test (xonly_ecdh_tag)
+const static auto BIP324_ECDH_TAG = ParseHex("4dbebba5394890bf7370e9ca2ca6112941b974c56950e5c4fef7f00a2bbf3e55");
```

Comment:
You can also use `const CHashWriter HASHER_BIP324_ECDH = TaggedHash("secp256k1_ellsq_xonly_ecdh");`, and then do the rest of the hashing using `(HashWriter(HASHER_BIP324_ECDH) << [rest of stuff] << [to hash]).GetSHA256()`.

Reply:
Super nice change - much cleaner. Thanks for showing me that. Done.